src/composables/info/feature-info.composable.ts

```diff
@@ -0,0 +1,413 @@
+            if (deltaX + deltaY < 6) {
+              singleclickEvent(evt)
+              startPixel.value = null
+              stopPixel.value = null
+            }
+          }
+        }, 500)
+      })()
+    })
+
+    listen(map, 'pointermove', event => {
+      const evt = event as MapBrowserEvent<any>
+      if (evt.dragging || isLoading.value) {
+        return
+      }
+      if (
+        drawStateActive.value /*TODO: || measureActive || streetviewActive*/
+      ) {
+        map.getViewport().style.cursor = ''
+      } else {
+        const pixel = map.getEventPixel(evt.originalEvent)
+        let hit: boolean | undefined = false
+        try {
+          hit = map.forEachLayerAtPixel(
+            pixel,
+            layer => {
+              if (layer) {
+                const metadata = layer.get('metadata')
+                if (
+                  metadata &&
+                  metadata['is_queryable'] &&
+                  layer.getVisible() &&
+                  layer.getOpacity() > 0
+                ) {
+                  return true
+                }
+              }
+              return false
+            },
+            {
+              layerFilter: layer => !!layer.getSource(),
+            }
+          )
+        } catch (error) {
+          hit = false
+        }
+        map.getViewport().style.cursor = hit ? 'pointer' : ''
+      }
+    })
+
+    watchEffect(() => {
+      if (fid.value) {
+        ;(async () => {
+          await getFeatureInfoById(fid.value as string)
+        })()
+      }
+    })
+  }
+
+  async function getFeatureInfoById(fid: string): Promise<void> {
+    const fids = fid.split(',')
+    for (const curFid of fids) {
+      const splittedFid = curFid.split('_')
+      const layersList = [splittedFid[0]]
+      const layerLabel: { [key: string]: string } = {}
+
+      try {
+        if (layersList.length > 0) {
+          setLoading(true)
+          map.getViewport().style.cursor = 'wait'
+
+          const url = new URL(INFO_SERVICE_URL)
+          url.searchParams.append('fid', curFid)
+          try {
+            const resp = await fetch(url.toString())
+
+            let openInfoPanel = false
+            if (screenSizeIsAtLeast('sm')) {
+              openInfoPanel = true
+              setInfoPanelHidden(false)
+            } else {
+              setInfoPanelHidden(true)
+            }
+
+            const node = findById(splittedFid[0])
+            if (node) {
+              layerLabel[splittedFid[0]] = node.layers as string
+            }
+
+            if (resp.ok) {
+              const data = await resp.json()
+              if (data.length > 0) {
+                showInfo(true, data, layerLabel, openInfoPanel, true)
+              }
+            }
+          } catch (error) {
+            reset()
+          }
+        }
+      } catch (error) {
+        reset()
+      }
+    }
+  }
+
+  async function singleclickEvent(evt: MapBrowserEvent<any>): Promise<void> {
```

Comment:
if i read the function name i don't know it will trigger an api call, either make it clear it the naming that they will be a call, or extract the api call into a new function (eg. existing services in  `src/services/api/` folder)

Reply:
Yep, renamed it and moved the API calls.